src/python/Utils/ExecuteDAO.py

```diff
@@ -0,0 +1,124 @@
+
+    usage:
+
+      For production:
+           source  /data/admin/wmagent/env.sh
+           source  /data/srv/wmagent/current/apps/wmagent/etc/profile.d/init.sh
+           python3 ExecuteDAO.py [--options]
+
+      For tier0:
+           source  /data/tier0/admin/env.sh
+           source  /data/tier0/srv/wmagent/current/apps/t0/etc/profile.d/init.sh
+           python3 ExecuteDAO.py [--options]
+    """)
+
+    parser.add_argument('-c', '--config', required=True,
+                        help="""The path to WMAGENT_CONFIG to be used, e.g.
+    for production: /data/srv/wmagent/current/config/wmagent/config.py
+    for tier0:      /data/tier0/srv/wmagent/current/config/tier0/config.py""")
+    parser.add_argument('-p', '--package', required=True,
+                        help='The package from which the DAO factory to be created for this execution, e.g. "WMCore.WMBS" or "WMComponent.DBS3Buffer"')
+    parser.add_argument('-m', '--module', required=True,
+                        help='The DAO Module to be executed, e.g. "Workflow.GetDeletableWorkflows" or "CountUndeletedBlocksByWorkflow"')
+    parser.add_argument('-d', '--debug', action='store_true', default=False,
+                        help='Set logging to debug mode.')
+    parser.add_argument('--dryRun', action='store_true', default=False,
+                        help='Simulation mode only')
+    args = parser.parse_args()
+
+    return args
+
+
+def loggerSetup(logLevel=logging.INFO):
+    """
+    Return a logger which writes everything to stdout.
+    """
+    logger = logging.getLogger(__name__)
+    outHandler = logging.StreamHandler(sys.stdout)
+    outHandler.setFormatter(logging.Formatter("%(asctime)s:%(levelname)s:%(module)s: %(message)s"))
+    outHandler.setLevel(logLevel)
+    logger.addHandler(outHandler)
+    logger.setLevel(logLevel)
+    return logger
+
+
+class ExecuteDAO():
+    """
+    A generic class to create the DAO Factory and execute the DAO module.
+    """
+    def __init__(self, logger, package=None, daoModule=None):
+        """
+        __init__
+        :param logger: The logger instance.
+        :param package: The Package from which the DAO factory to be initialised.
+        :param module: The DAO module to be executed.
+        """
+        # Connecting to DB
+        myThread = threading.currentThread()
+        connectToDB()
+        self.logger = logger
+        self.dbi = myThread.dbi
+        self.package = package
+        self.daoModule = daoModule
+
+        self.daoFactory = DAOFactory(package=package,
+                                     logger=self.logger,
+                                     dbinterface=self.dbi)
+        self.logger.info("DAO Factory initialised from package: %s", self.package)
+
+        self.dao = self.daoFactory(classname=daoModule)
+        self.logger.info("DAO Module initialised as: %s", self.daoModule)
+
+    def __call__(self):
```

Comment:
And a basic docstring here as well, even though it's quite clear what will happen :)

Reply:
done

Reply:
I am sure you could have done a much better job, but okay.